### PR TITLE
Change Tab and FormTab components to function components.

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-ui-materialui/src/detail/Tab.js [hunk 1]
```diff
@@ -58,8 +58,9 @@ const sanitizeRestProps = ({
  *     );
  *     export default App;
  */
-class Tab extends Component {
-    renderHeader = ({ className, label, icon, value, translate, ...rest }) => (
+const Tab = props => {
+    const translate = useTranslate();
+    const renderHeader = ({ className, label, icon, value, ...rest }) => (
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/detail/Tab.js]: This function is parameters and variables in the closure. I think you should decide between a function without parameters (and dereferencing all props in the Tab component signature), or a full-fledged component (in which case no parameter can be defined outside, as you did with translate). I'd personally opt for the first option.

File: packages/ra-ui-materialui/src/form/FormTab.js [hunk 2]
```diff
@@ -18,8 +18,9 @@ const sanitizeRestProps = ({
 
 const hiddenStyle = { display: 'none' };
 
-class FormTab extends Component {
-    renderHeader = ({ className, label, icon, value, translate, ...rest }) => {
+const FormTab = props => {
+    const translate = useTranslate();
+    const renderHeader = ({ className, label, icon, value, ...rest }) => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.js]: same remark

File: packages/ra-ui-materialui/src/detail/Tab.js (hunk 1 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/detail/Tab.js]: Ok, Done.

File: packages/ra-ui-materialui/src/form/FormTab.js (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/FormTab.js]: Ok, Done.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**